File: 02.ProgrammingFundametals-C#/02.Lists/13.ArrayManipulator/ArrayManipulator.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class ArrayManipulator
{
    public static void Main()
    {
        var inputNums = Console.ReadLine()
            .Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries)
            .Select(int.Parse)
            .ToList();

        string commands = Console.ReadLine();
        while (commands != "print")
        {
            string[] tokens = commands.Split(' ');
            string command = tokens[0];
            switch (command)
            {
                case "add":
                    int index = int.Parse(tokens[1]);
                    int element = int.Parse(tokens[2]);
                    inputNums.Insert(index, element);

                    break;
                case "addMany":
                    inputNums.InsertRange(int.Parse(tokens[1]), tokens.Skip(2)
                        .Select(int.Parse)
                        .ToArray());
                    break;
                case "contains":
                    int containsElement = int.Parse(tokens[1]);
                    Console.WriteLine(inputNums.IndexOf(containsElement));
                    
                    break;
                case "remove":
                    int indexForRemove = int.Parse(tokens[1]);
                    inputNums.RemoveAt(indexForRemove);
                    break;
                case "shift":
                    int position = int.Parse(tokens[1]);
                    position %= inputNums.Count;
                    var rem = inputNums.Take(position)
                        .ToList();
                    inputNums.RemoveRange(0, position);
                    inputNums.AddRange(rem);
                    break;
                case "sumPairs":
                    List<int> sumPairs = new List<int>();
                    for (int i = 0; i < inputNums.Count - 1; i++)
                    {
                        int sum = inputNums[i] + inputNums[i + 1];
                        inputNums[i] = sum;
                        inputNums.RemoveAt(i + 1);
                    }
                    if (inputNums.Count % 2 != 0)
                    {
                        sumPairs.Add(inputNums[inputNums.Count - 1]);
                    }
                    break;
            }
            commands = Console.ReadLine();
        }
        string result = string.Join(", ", inputNums);
        Console.WriteLine($"[{result}]");
    }
}

File: 07.TechFundamentalsWithJS/13.AdvancedArrayExercise/08.arrayManipulator/arrayManipulator.js
function arrayManipulator(params, args) {
    let array = params.slice();
    for (const line of args) {
        let parts = line.split(/\s+/);
        let command = parts.shift();
        let index;

        switch (command) {
            case 'add':
                index = +parts.shift();
                let element = +parts.shift();
                array.splice(index, 0, element);
                break;

            case 'addMany':
                index = +parts.shift();
                let elements = parts.map(Number).slice();
                array.splice(index, 0, ...elements);
                break;
            case 'contains':
                let searchElement = +parts;
                console.log(array.indexOf(searchElement));
                break;

            case 'remove':
                array.splice(+parts, 1);
                break;

            case 'shift':
                let position = +parts % array.length;
                for (let i = 0; i < position; i++) {
                    let number = array.shift();
                    array.push(number);
                }
                break;

            case 'sumPairs':
                let newArr = [];
                for (let i = 0; i < array.length - 1; i += 2) {
                    let curr = array[i];
                    let next = array[1 + i];
                    newArr.push(curr + next);
                }

                if (array.length % 2 != 0) {
                    newArr.push(array.pop());
                }
                array = newArr.slice();
                break;

            case 'print':
                return (`[ ${array.join(', ')} ]`);
        }
    }
}

const output = arrayManipulator(
    [1, 2, 3, 4, 5],
    ['addMany 5 9 8 7 6 5', 'contains 15', 'remove 3', 'shift 1', 'print']
);

console.log(output);
// [ 2, 3, 5, 9, 8, 7, 6, 5, 1 ]